src/wallet/walletdb.cpp

```diff
@@ -286,7 +287,12 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             // LoadToWallet call below creates a new CWalletTx that fill_wtx
             // callback fills with transaction metadata.
             auto fill_wtx = [&](CWalletTx& wtx, bool new_tx) {
-                assert(new_tx);
+                if (!new_tx) {
+                    // There's probably some corruption here since the tx we just tried to load was already in the wallet
+                    // This error is recoverable with zapwallettxs and is not a major failure
```

Comment:
Isn't zapwallettxs about to be removed?

Reply:
I guess here's an argument to keep it.

The question is really whether it is possible to hit this error in normal operation.

Reply:
The test #19078 is hitting it. Does salvagewallet count as "normal operation"? If not, it should probably be removed.

Reply:
Looking at it further, it is surprising to me that #19078 hits this error because it shouldn't have any transactions left after the salvage. Salvage is supposed to only extract keys. This indicates that salvage has an additional bug somewhere.

Reply:
Oh, the error is because `ReadKeyValue` is called by salvage so it's happening during the salvage, rather than during the loading of the wallet post-salvage.

Reply:
#19805 is a fix to #19078's failure that sidesteps this problem entirely.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/19793#discussion_r476588885

> #19805 is a fix to #19078's failure that sidesteps this problem entirely.

#19805 seems like a better way to fix the test failure. In case we want to merge this PR as well to remove the bad assert https://github.com/bitcoin/bitcoin/pull/19793#pullrequestreview-474262517, I updated the comment, just changing "There's probably some corruption" to "There's some corruption"